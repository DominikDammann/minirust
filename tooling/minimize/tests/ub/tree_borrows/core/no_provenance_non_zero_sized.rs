//@ compile-flags: --minimize-tree-borrows

// Check that we forbid retagging a non-zero-sized pointer without provenance.

#![feature(strict_provenance)]

fn foo(_x: &mut u8) {
    // UB! Retagging a non-zero-sized pointer without provenance.
}

fn main() {
    let xraw = &mut 42u8 as *mut u8;
    let ptr = std::ptr::without_provenance_mut::<u8>(xraw.addr());
    foo(unsafe { &mut *ptr });
}
