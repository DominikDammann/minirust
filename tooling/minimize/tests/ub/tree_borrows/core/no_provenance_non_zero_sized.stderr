fatal error: UB: Tree Borrows: non-zero-sized reborrow of a pointer without provenance
